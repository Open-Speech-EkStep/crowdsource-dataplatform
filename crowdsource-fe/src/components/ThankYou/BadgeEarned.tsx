import Image from 'next/image';

import ContributionDetails from 'components/ContributionDetails';
import Link from 'components/Link';
import SocialShareIcons from 'components/SocialShareIcons';
import TwoColumn from 'components/TwoColumn';
import { capitalizeFirstLetter } from 'utils/utils';

import styles from './ThankYou.module.scss';

const BadgeImage = () => (
  <div className={`${styles.medal} mx-auto`}>
    <Image
      src={`/images/en/badges/en_bolo_bronze_contribute.svg`}
      width="140"
      height="180"
      alt={`Bronze Badge english`}
    />
  </div>
);

const ShareOn = () => (
  <div
    className={`${styles.socialShare} d-inline-flex align-items-center border border-1 border-primary px-4`}
  >
    <span className="me-3">Share on</span>
    <SocialShareIcons />
  </div>
);

const DownloadAndShare = () => (
  <div className="d-flex flex-column align-items-center flex-md-row align-items-md-baseline">
    <Link href="http://test.com">
      <a
        className={`${styles.download} d-inline-flex align-items-center border border-1 border-primary px-4`}
      >
        Download
        <span className="d-flex ms-2">
          <Image src="/images/download_icon.svg" width="12" height="15" alt="download-image" />
        </span>
      </a>
    </Link>
    <div className="mt-4 mt-md-0 ms-md-4">
      <ShareOn />
    </div>
  </div>
);

interface BadgeEarnedProps {
  initiative: string;
  badgeType: string;
  contributionCount: number;
  pageMediaTypeStr: string;
  language: string;
}

const BadgeEarned = ({
  initiative,
  badgeType,
  contributionCount,
  pageMediaTypeStr,
  language,
}: BadgeEarnedProps) => {
  return (
    <ContributionDetails
      top={
        <TwoColumn
          left={<BadgeImage />}
          right={
            <div className="text-center text-md-start">
              <h4 className="px-8 px-md-0">
                Youâ€™ve earned your {capitalizeFirstLetter(initiative)} India{' '}
                <span className="text-strong-warning">{badgeType} Bhasha Samarthak</span> badge.
              </h4>
              <p className="display-3 mt-5 mt-6">
                Transcribed{' '}
                <strong>
                  {contributionCount} {pageMediaTypeStr}
                </strong>{' '}
                in {language}
              </p>
            </div>
          }
          withSep={false}
        />
      }
      bottom={<DownloadAndShare />}
    />
  );
};

export default BadgeEarned;
