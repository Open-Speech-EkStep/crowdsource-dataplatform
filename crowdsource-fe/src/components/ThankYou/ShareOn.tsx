import { useTranslation } from 'next-i18next';
import { useRouter } from 'next/router';

import SocialShareIcons from 'components/SocialShareIcons';
import nodeConfig from 'constants/nodeConfig';
import { capitalizeFirstLetter } from 'utils/utils';

import styles from './ThankYou.module.scss';

interface ShareOnProps {
  initiativeName: string;
  rank: Number;
  language: string;
}

const ShareOn = ({ initiativeName, rank, language }: ShareOnProps) => {
  const { locale } = useRouter();
  const { t } = useTranslation();
  const socialSharingTextWithRank = t('socialSharingTextWithRank', {
    rank: rank,
    initiativeName: initiativeName,
    language: capitalizeFirstLetter(language),
    url: `${nodeConfig.appUrl}${nodeConfig.contextRoot}/${locale}/home`,
  });

  return (
    <div
      className={`${styles.socialShare} d-inline-flex align-items-center border border-1 border-primary px-4`}
    >
      <span className="me-3">{t('shareOn')}</span>
      <SocialShareIcons socialShareText={socialSharingTextWithRank} />
    </div>
  );
};

export default ShareOn;
