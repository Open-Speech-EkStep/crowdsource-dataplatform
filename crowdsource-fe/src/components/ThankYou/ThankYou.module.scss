.root {
  max-width: 50rem;
  position: relative;
  z-index: 1;
}

.medal {
  width: 5.875rem;

  @include media-breakpoint-up(md) {
    width: 8.75rem;
  }
}

.download {
  border-radius: 20px;
  box-shadow: 0 4px 12px rgba(15, 39, 73, 0.2);
  color: var(--primary-color);
  font-size: 0.875rem;
  height: 2.25rem;

  &:hover {
    color: var(--primary-color);
  }
}

.socialShare {
  border-radius: 20px;
  box-shadow: 0 4px 12px rgba(15, 39, 73, 0.2);
  color: var(--primary-color);
  font-size: 0.875rem;
  height: 2.25rem;

  &:hover {
    color: var(--primary-color);
  }
}

.badgeBg {
  background-color: var(--badge-bg-color);
  border-radius: 20px;
}

.count {
  color: var(--link-color);
}

.bar {
  @include media-breakpoint-up(md) {
    bottom: 1.5rem;
    position: sticky;
    z-index: 1;
  }
}
