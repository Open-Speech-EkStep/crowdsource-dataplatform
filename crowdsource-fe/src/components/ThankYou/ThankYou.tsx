import React, { Fragment, useEffect, useState } from 'react';

import { Trans, useTranslation } from 'next-i18next';
import Image from 'next/image';
import { useRouter } from 'next/router';

import CompareLanguages from 'components/CompareLanguages';
import ContributeMore from 'components/ContributeMore';
import ContributionDetails from 'components/ContributionDetails';
import TwoColumn from 'components/TwoColumn';
import TyTargetProgress from 'components/TyTargetProgress';
import apiPaths from 'constants/apiPaths';
import {
  CONTRIBUTION_MAPPING,
  INITIATIVES_MAPPING,
  INITIATIVES_MEDIA_MAPPING,
} from 'constants/initiativeConstants';
import { LOCALE_LANGUAGES } from 'constants/localesConstants';
import localStorageConstants from 'constants/localStorageConstants';
import {
  pageSourceConstants,
  pageSourceConstants2,
  pageSourceConstants3,
  pageSourceConstants4,
} from 'constants/pageRouteConstants';
import routePaths from 'constants/routePaths';
import { useFetchWithInit } from 'hooks/useFetch';
import useLocalStorage from 'hooks/useLocalStorage';
import type { Initiative } from 'types/Initiatives';
import type SpeakerDetails from 'types/SpeakerDetails';
import type { ThankYouReward } from 'types/ThankYou';
import { capitalizeFirstLetter } from 'utils/utils';

import styles from './ThankYou.module.scss';

import { BadgeEarned, ShareOn } from '.';

const YourBadge = (props: any) => {
  const route = useRouter();
  const { t } = useTranslation();

  const [contributionLanguage] = useLocalStorage<string>(localStorageConstants.contributionLanguage);
  const currentContributionAlias = LOCALE_LANGUAGES[contributionLanguage ?? ''];

  return (
    <div className="text-center">
      <h5 className="fw-light mb-3">{t('yourBadge')}</h5>
      <div className={`${styles.badgeBg} p-3`}>
        <Image
          src={`/images/${currentContributionAlias}/badges/${currentContributionAlias}_${
            props.initiative
          }_${props.badgeType.toLowerCase()}_${props.source}.svg`}
          width="94"
          height="120"
          alt={`${props.badgeType} badge`}
        />
      </div>
      <span className="d-flex justify-content-center mt-3 display-5 font-family-rowdies fw-light">
        <Trans
          i18nKey={`${pageSourceConstants4[route.asPath]}Acheived`}
          defaults={`${pageSourceConstants4[route.asPath]}Acheived`}
          values={{
            badgeType: capitalizeFirstLetter(t(props.badgeType.toLowerCase())),
          }}
          components={{ span: <span className="text-strong-warning ms-1 me-1" /> }}
        />
      </span>
    </div>
  );
};

interface ThankYouProps {
  initiative: Initiative;
}

const ThankYou = ({ initiative }: ThankYouProps) => {
  const route = useRouter();
  const { t } = useTranslation();

  const [mode, setMode] = useState<string>('see');

  const [contributionLanguage] = useLocalStorage<string>(localStorageConstants.contributionLanguage);

  const [translatedLanguage] = useLocalStorage<string>(localStorageConstants.translatedLanguage);

  const [speakerDetails] = useLocalStorage<SpeakerDetails>(localStorageConstants.speakerDetails);

  const { data: rewardData, mutate: rewardMutate } = useFetchWithInit<ThankYouReward>(
    `${apiPaths.rewards}?type=${
      INITIATIVES_MEDIA_MAPPING[initiative]
    }&language=${contributionLanguage}&source=${pageSourceConstants[route.asPath]}&userName=${
      speakerDetails?.userName
    }`,
    {
      revalidateOnMount: false,
    }
  );

  const translatedContributionLanguage = `${t(`${contributionLanguage?.toLowerCase()}`)}${
    initiative === INITIATIVES_MAPPING.likho ? `-${t(`${translatedLanguage?.toLowerCase()}`)}` : ''
  }`;

  useEffect(() => {
    if (contributionLanguage && speakerDetails) {
      rewardMutate();
    }
  }, [contributionLanguage, rewardMutate, speakerDetails]);

  return (
    <Fragment>
      <div className={`${styles.root} mx-auto`} data-testid="ThankYou">
        {rewardData?.isNewBadge && <h2 className="text-center">{t('congratulationText')}</h2>}
        {rewardData?.badges && rewardData?.badges.length === 0 && rewardData?.contributionCount === 0 && (
          <h2 className="text-center">{t('contributeYourLanguage')}</h2>
        )}
        {!rewardData?.isNewBadge && rewardData?.contributionCount !== 0 && (
          <h2 className="text-center">
            <Trans
              i18nKey={`${INITIATIVES_MEDIA_MAPPING[initiative]}${capitalizeFirstLetter(
                pageSourceConstants2[route.asPath]
              )}Message`}
              defaults={`${INITIATIVES_MEDIA_MAPPING[initiative]}${capitalizeFirstLetter(
                pageSourceConstants2[route.asPath]
              )}Message`}
              values={{
                count: rewardData?.contributionCount,
              }}
              components={{ span: <span className={styles.count} /> }}
            />
          </h2>
        )}
        <section className="mt-8">
          <TyTargetProgress
            initiative={INITIATIVES_MAPPING[initiative]}
            initiativeType={INITIATIVES_MEDIA_MAPPING[initiative]}
            source={pageSourceConstants[route.asPath] || ''}
            language={contributionLanguage ?? ''}
          />
        </section>
        {rewardData?.isNewBadge && (
          <section className="mt-8">
            <BadgeEarned
              initiative={initiative}
              badgeType={rewardData?.currentBadgeType.toLowerCase()}
              contributionCount={rewardData?.contributionCount}
              language={translatedContributionLanguage}
              source={pageSourceConstants[route.asPath]}
              winningBadge={rewardData?.badges[rewardData?.badges.length - 1]}
            />
            <div className="mt-8 mt-md-9">
              <TwoColumn
                left={
                  <div className="d-md-flex text-center text-md-start h-100 align-items-md-center">
                    <h4>
                      {t(`${mode}ParticipationText`, {
                        language: translatedContributionLanguage,
                      })}
                    </h4>
                  </div>
                }
                right={
                  <CompareLanguages
                    initiative={initiative}
                    graphLabel={t(
                      `${INITIATIVES_MEDIA_MAPPING[initiative]}${capitalizeFirstLetter(
                        pageSourceConstants3[route.asPath]
                      )}GraphYLabel3`
                    )}
                    dataBindigValue={CONTRIBUTION_MAPPING[route.asPath]}
                    isTopLanguage={setMode}
                    graphHeading={t(`${mode}ParticipationText`, {
                      language: translatedContributionLanguage,
                    })}
                    showHeader={false}
                    source={pageSourceConstants[route.asPath]}
                  />
                }
              />
            </div>
          </section>
        )}
        {rewardData?.badges && rewardData?.badges.length === 0 && (
          <section className="mt-8">
            <ContributionDetails
              top={
                <TwoColumn
                  left={
                    <div className="d-md-flex text-center text-md-start h-100 align-items-md-center">
                      <h4>
                        {t(`${mode}ParticipationText`, {
                          language: translatedContributionLanguage,
                        })}
                      </h4>
                    </div>
                  }
                  right={
                    <CompareLanguages
                      initiative={initiative}
                      graphLabel={t(
                        `${INITIATIVES_MEDIA_MAPPING[initiative]}${capitalizeFirstLetter(
                          pageSourceConstants3[route.asPath]
                        )}GraphYLabel3`
                      )}
                      dataBindigValue={CONTRIBUTION_MAPPING[route.asPath]}
                      isTopLanguage={setMode}
                      graphHeading={t(`${mode}ParticipationText`, {
                        language: translatedContributionLanguage,
                      })}
                      showHeader={false}
                      source={pageSourceConstants[route.asPath]}
                    />
                  }
                />
              }
              bottom={<ShareOn />}
            />
          </section>
        )}
        {rewardData?.badges && rewardData?.badges.length !== 0 && !rewardData.isNewBadge && (
          <section className="mt-8">
            <ContributionDetails
              top={
                <TwoColumn
                  left={
                    <YourBadge
                      badgeType={rewardData?.currentBadgeType}
                      initiative={INITIATIVES_MAPPING[initiative]}
                      source={pageSourceConstants[route.asPath]}
                    />
                  }
                  right={
                    <CompareLanguages
                      initiative={initiative}
                      graphLabel={t(
                        `${INITIATIVES_MEDIA_MAPPING[initiative]}${capitalizeFirstLetter(
                          pageSourceConstants3[route.asPath]
                        )}GraphYLabel3`
                      )}
                      dataBindigValue={CONTRIBUTION_MAPPING[route.asPath]}
                      isTopLanguage={setMode}
                      graphHeading={t(`${mode}ParticipationText`, {
                        language: translatedContributionLanguage,
                      })}
                      showHeader={true}
                      source={pageSourceConstants[route.asPath]}
                    />
                  }
                />
              }
              bottom={<ShareOn />}
            />
          </section>
        )}
      </div>
      <section className={`${styles.bar} mt-8 mt-md-12`}>
        <ContributeMore
          initiative={initiative}
          source={pageSourceConstants[route.asPath]}
          nextMileStone={rewardData?.nextMilestone ?? 0}
          contributionCount={rewardData?.contributionCount ?? 0}
          nextBadgeType={rewardData?.nextBadgeType ?? ''}
          url={`${
            routePaths[`${initiative}India${capitalizeFirstLetter(pageSourceConstants[route.asPath])}`]
          }`}
          badges={rewardData?.badges}
          isTopLanguage={mode}
          badgeType={rewardData?.currentBadgeType}
        />
      </section>
    </Fragment>
  );
};

export default ThankYou;
