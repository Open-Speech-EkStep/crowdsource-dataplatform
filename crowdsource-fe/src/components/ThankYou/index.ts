export { default } from './ThankYou';
export { default as BadgeEarned } from './BadgeEarned';
export { default as ShareOn } from './ShareOn';
