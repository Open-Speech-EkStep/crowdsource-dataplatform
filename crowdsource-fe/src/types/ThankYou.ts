export interface ThankYouReward {
  badgeId: string;
  badges: Array<any>[];
  contributionCount: number;
  currentBadgeType: string;
  isNewBadge: false;
  nextBadgeType: string;
  nextMilestone: number;
}
