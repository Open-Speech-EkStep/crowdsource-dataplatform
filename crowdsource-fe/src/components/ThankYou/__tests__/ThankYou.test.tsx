import { when } from 'jest-when';
import router from 'next/router';

import { render, waitFor, screen } from 'utils/testUtils';

import ThankYou from '../ThankYou';

describe('ThankYou', () => {
  const setup = async (initiative: 'tts' | 'ocr' | 'asr' | 'translation', data: any) => {
    router.asPath = '/tts-initiative/contribute/thank-you';
    const speakerDetails = {
      userName: 'abc',
      motherTongue: '',
      age: '',
      gender: '',
      language: 'Hindi',
      toLanguage: '',
    };

    when(localStorage.getItem)
      .calledWith('speakerDetails')
      .mockImplementation(() => JSON.stringify(speakerDetails));

    const language = 'English';

    when(localStorage.getItem)
      .calledWith('contributionLanguage')
      .mockImplementation(() => language);

    when(localStorage.getItem)
      .calledWith('translatedLanguage')
      .mockImplementation(() => 'Hindi');

    fetchMock
      .doMockIf('/rewards?type=asr&language=English&source=contribute&userName=abc')
      .mockResponse(JSON.stringify(data));

    const renderResult = render(<ThankYou initiative={initiative} />);

    await waitFor(() => {
      expect(localStorage.getItem).toBeCalledWith('contributionLanguage');
    });
    await waitFor(() => {
      expect(localStorage.getItem).toBeCalledWith('speakerDetails');
    });

    await waitFor(() => {
      expect(fetchMock).toBeCalledWith('/rewards?type=asr&language=English&source=contribute&userName=abc', {
        credentials: 'include',
        method: 'GET',
        mode: 'cors',
      });
    });

    return renderResult;
  };

  // verifyAxeTest(setup());

  it('should render the component and matches it against stored snapshot', async () => {
    const data = {
      badgeId: '',
      badges: [],
      contributionCount: 0,
      currentBadgeType: '',
      isNewBadge: false,
      nextBadgeType: 'Bronze',
      nextMilestone: 5,
    };
    const { asFragment } = await setup('tts', data);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should test when no badge earned', async () => {
    const data = {
      badgeId: '',
      badges: [],
      contributionCount: 0,
      currentBadgeType: '',
      isNewBadge: false,
      nextBadgeType: 'Bronze',
      nextMilestone: 5,
    };
    await setup('tts', data);

    expect(screen.getByText('contributeYourLanguage')).toBeInTheDocument();

    expect(screen.getByText('seeParticipationText')).toBeInTheDocument();
  });

  it('should test when user earned the badge', async () => {
    const data = {
      badgeId: '',
      badges: [
        {
          generated_badge_id: 'de68ee51-d018-4b20-ba90-1d608bd35eda',
          grade: 'Bronze',
        },
      ],
      contributionCount: 5,
      currentBadgeType: 'Bronze',
      isNewBadge: true,
      nextBadgeType: 'Bronze',
      nextMilestone: 50,
    };
    await setup('tts', data);

    expect(screen.getByText('congratulationText')).toBeInTheDocument();

    expect(screen.getByText('ttsBadgeEarnedText')).toBeInTheDocument();

    expect(screen.getByText('seeParticipationText')).toBeInTheDocument();
  });

  it('should test after user earned the badge', async () => {
    const data = {
      badgeId: '',
      badges: [
        {
          generated_badge_id: 'de68ee51-d018-4b20-ba90-1d608bd35eda',
          grade: 'Bronze',
        },
      ],
      contributionCount: 6,
      currentBadgeType: 'Bronze',
      isNewBadge: false,
      nextBadgeType: 'Bronze',
      nextMilestone: 50,
    };
    await setup('tts', data);

    expect(screen.getByText('ttsContributedMessage')).toBeInTheDocument();

    expect(screen.getByText('yourBadge')).toBeInTheDocument();

    expect(screen.getByText('contributorAcheived')).toBeInTheDocument();
  });

  it('should test after user earned the silver badge', async () => {
    const data = {
      badgeId: '',
      badges: [
        {
          generated_badge_id: 'de68ee51-d018-4b20-ba90-1d608bd35eda',
          grade: 'Bronze',
        },
        {
          generated_badge_id: 'de68ee51-d018-4b20-ba90-1d608bd35eda',
          grade: 'Silver',
        },
      ],
      contributionCount: 60,
      currentBadgeType: 'Silver',
      isNewBadge: false,
      nextBadgeType: 'Bronze',
      nextMilestone: 100,
    };
    await setup('tts', data);

    expect(screen.getByText('ttsContributedMessage')).toBeInTheDocument();

    expect(screen.getByText('yourBadge')).toBeInTheDocument();

    expect(screen.getByText('contributorAcheived')).toBeInTheDocument();
  });
});
