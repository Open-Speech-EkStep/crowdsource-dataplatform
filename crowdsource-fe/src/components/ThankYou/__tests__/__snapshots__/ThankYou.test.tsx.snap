// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThankYou should render the component and matches it against stored snapshot 1`] = `
<DocumentFragment>
  <div
    class="root mx-auto"
    data-testid="ThankYou"
  >
    <h2
      class="text-center"
    >
      contributeYourLanguage
    </h2>
    <section
      class="mt-8"
    />
    <section
      class="mt-8"
    >
      <div
        class="root rounded-20"
        data-testid="badgeDescriptionRow"
      >
        <div
          class="rounded-20 bg-light p-5"
        >
          <div>
            <div
              class="row"
            >
              <div
                class="pb-4 pb-md-0 col-md-4 col-12"
              >
                <div
                  class="d-md-flex text-center text-md-start h-100 align-items-md-center"
                >
                  <h4>
                    seeParticipationText
                  </h4>
                </div>
              </div>
              <div
                class="col-md-auto col-12"
              >
                <span
                  class="d-flex h-100 sep"
                />
              </div>
              <div
                class="pt-4 pt-md-0 col"
              />
            </div>
          </div>
          <div
            class="mt-4 text-center text-md-start"
          >
            <div
              class="socialShare d-inline-flex align-items-center border border-1 border-primary px-4"
            >
              <span
                class="me-3"
              >
                shareOn
              </span>
              <a
                class="socialShareLink mx-2 d-inline-block ms-0"
                href="https://www.facebook.com/sharer/sharer.php?u=/en/home&quote=socialSharingTextWithRank"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Share on Facebook"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="/images/fb-icon.svg"
                    srcset="/images/fb-icon.svg 1x, /images/fb-icon.svg 2x"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </a>
              <a
                class="socialShareLink mx-2 d-inline-block"
                href="https://www.linkedin.com/shareArticle?mini=true&url=/en/home&title=metaOGTitle&summary=socialSharingTextWithRank"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Share on LinkedIn"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="/images/linkedin-icon.svg"
                    srcset="/images/linkedin-icon.svg 1x, /images/linkedin-icon.svg 2x"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </a>
              <a
                class="socialShareLink mx-2 d-inline-block"
                href="https://twitter.com/intent/tweet?text=socialSharingTextWithRank"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Share on Twitter"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="/images/twitter-icon.svg"
                    srcset="/images/twitter-icon.svg 1x, /images/twitter-icon.svg 2x"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </a>
              <a
                class="socialShareLink mx-2 d-inline-block me-0"
                href="https://api.whatsapp.com/send?text=socialSharingTextWithRank"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Share on Whatsapp"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="/images/whatsapp.svg"
                    srcset="/images/whatsapp.svg 1x, /images/whatsapp.svg 2x"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
  <section
    class="bar mt-8 mt-md-12"
  >
    <div
      class="root d-flex align-items-md-center flex-column flex-md-row mx-auto justify-content-md-between p-5 py-md-4 px-md-5 bg-light text-center text-md-start"
    >
      <div
        class="d-flex flex-column flex-md-row align-items-md-center"
      >
        <div
          class="d-flex justify-content-center"
        />
        <p
          class="text mt-3 mt-md-0 ms-md-3 display-4"
        >
          ttsContributeRemainingText
        </p>
        <div
          class="disabled d-flex justify-content-center mt-4 mt-md-0 mb-3 mb-md-0"
        >
          <div
            class="medal upcomingBadge  d-flex mx-2 flex-shrink-0"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2748%27%20height=%2760%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Bronze Badge"
                data-nimg="intrinsic"
                decoding="async"
                src="/images/vakyansh/en/badges/en_tts_bronze_contribute.svg"
                srcset="/images/vakyansh/en/badges/en_tts_bronze_contribute.svg 1x, /images/vakyansh/en/badges/en_tts_bronze_contribute.svg 2x"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="medal otherBadges  d-flex mx-2 flex-shrink-0"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2748%27%20height=%2760%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Bronze Badge"
                data-nimg="intrinsic"
                decoding="async"
                src="/images/vakyansh/en/badges/en_tts_silver_contribute.svg"
                srcset="/images/vakyansh/en/badges/en_tts_silver_contribute.svg 1x, /images/vakyansh/en/badges/en_tts_silver_contribute.svg 2x"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="medal otherBadges  d-flex mx-2 flex-shrink-0"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2748%27%20height=%2760%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Bronze Badge"
                data-nimg="intrinsic"
                decoding="async"
                src="/images/vakyansh/en/badges/en_tts_gold_contribute.svg"
                srcset="/images/vakyansh/en/badges/en_tts_gold_contribute.svg 1x, /images/vakyansh/en/badges/en_tts_gold_contribute.svg 2x"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="medal otherBadges  d-flex mx-2 flex-shrink-0"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2748%27%20height=%2760%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Bronze Badge"
                data-nimg="intrinsic"
                decoding="async"
                src="/images/vakyansh/en/badges/en_tts_platinum_contribute.svg"
                srcset="/images/vakyansh/en/badges/en_tts_platinum_contribute.svg 1x, /images/vakyansh/en/badges/en_tts_platinum_contribute.svg 2x"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
        </div>
        <a
          class="text mt-4 mt-md-0 display-5 ms-md-3 flex-shrink-0"
          href="/badge-detail?initiative=asr&language=English&source=contribute&badge=bronze"
        >
          <b>
            knowMore
          </b>
        </a>
      </div>
      <button
        class="mt-8 mt-md-0 ms-md-3 info primary btn btn-primary"
        data-testid="Button"
        type="button"
      >
        contributeMore
      </button>
    </div>
  </section>
</DocumentFragment>
`;
